CMakeLists.txt

```diff
@@ -28,6 +28,18 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     add_definitions(-DMACOSX)
 endif()
 
+if (USE_BALTIMORE_CERT)
+    add_definitions(-DUSE_BALTIMORE_CERT)
+endif()
+
+if (USE_MICROSOFTAZURE_DE_CERT)
+    add_definitions(-DUSE_MICROSOFTAZURE_DE_CERT)
+endif()
+
+if (USE_PORTAL_AZURE_CN_CERT)
+    add_definitions(-DUSE_PORTAL_AZURE_CN_CERT)
+endif()
+
```

Comment:
It's close, but …

Look inside the CMakeLists.txt for "use_uamqp".

The option should be defined like that.

option(use_baltimore_cert "set use_baltimore_cert to ON if the Baltimore cert is to be used, set to OFF to not use it" OFF)

And on the if, it should be checked if its value is ON.

if (${use_baltimore_cert}) 
    add_definitions(-DUSE_BALTIMORE_CERT) 
endif()

------

The same above needs to be done for all 3 new variables.

-----

And down where you call cmake, use -Duse_baltimore_cert=ON (notice the lowercase).

Reply:
Thanks for the specificity!